### DISCUSSION TITLE
Yjs extension cursor text and styling

### DISCUSSION THREAD

**Discussion by anotheri** [remirror/remirror#857] [Q&A] [answered]
@ifiokjr hey there, would you mind helping me with how to set a username for cursor? and how can I set class name to change its styling?

instead of showing something like this (I guess this comes from Yjs awareness, but not sure)?
![image](https://user-images.githubusercontent.com/885799/107380934-5f1ef080-6aff-11eb-9334-8aea56d9bff6.png)

it renders cursor for a whole line although cursor setup in the end of "123" text?
i guess I can rewrite `.ProseMirror-yjs-cursor` styles but wondering if there is any better way for that.

**Comment by ifiokjr** [comment]
@anotheri I'll have a play with `yjs` this weekend and let you know my findings.

I haven't had to use it in any serious way in the past so I'm not the best for how to customise the integration.

**Reply by anotheri** [reply]
i.e. it comes from Yjs (provider/implmentation), and not from yjs-extension per se, correct?

**Comment by anotheri** [comment, marked as answer]
Ok, so I've found the solution here: https://github.com/yjs/y-prosemirror#remote-cursors.
**Key point: The user name should be set up on the client-side!** (Although I believe the auth on WebSockets I've implemented is a good move anyway.)

I'm using this react helper component to get the Yjs extension for Remirror. This is optional functionality on Remirror editor in our project, so I keep it decoupled and include it in the editor whenever we need it.
 
 ```
const YjsExt = ({ room, user } = {}) => {
  const getProvider = useCallback(() => {
    if (room && user) {
      const doc = new Doc();
      
      // const provider = new WebrtcProvider(room, doc);
      
      const socketUrl = getSocketUrl({ path: '' });
      const provider = new WebsocketProvider(socketUrl, room, doc, {
        params: getSocketAuthParams(user)
      });
      
      // Here you define name which will be shown on remirror cursor
      provider.awareness.setLocalStateField('user', {
        name: user.name,
        email: user.email
      });

      return provider;
    }

    return null;
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [room, user]);

  if (!room || !user) {
    return null;
  }

  return new YjsExtension({ getProvider });
};
 ```
 
Hopefully, it will save some time for you. 🙂

**Reply by ifiokjr** [reply]
Brilliant! Thanks for sharing your findings 👍
